### PR TITLE
Create MVP book recommendation feature

### PR DESCRIPTION
**Pull request by kashiish**
These commits implement the book recommendation feature for the MVP. When a user searches for a topic, there will be two tabs, one for tutors and one for books. The books page will display a list of books relevant to the searched topic. I created a servlet called BookServlet that acts as a mock API to get a list of books for a searched topic. This will be replaced by the Google Books API in the future. I also updated and added Jasmine tests for creating a book result element and switching between tabs.

### CODE REVIEW

File: src/main/webapp/search.js [hunk 1]
```diff
@@ -81,7 +81,8 @@ async function getBooks(topic) {
             return;
         }
 
-        numSearchResults.innerText = "Found " + results.length + " books for " + topic;
+        //Only make "books" plural if there are 0 or more than 1 books
+        numSearchResults.innerText = "Found " + results.length + (results.length > 1 || results.length === 0 ? " books for " : " book for ") + topic;
```
Comment (samfalberg) [inline src/main/webapp/search.js:85]: lol good catch

File: src/main/java/com/google/sps/data/Book.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.lang.String;
+
```
Comment (betrevisan) [inline src/main/java/com/google/sps/data/Book.java:18]: Anand recommended that we added javadocs to these. This is what I followed to do that: https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required

File: src/main/webapp/search.js [hunk 3]
```diff
@@ -43,31 +56,95 @@ async function getSearchResultsHelper(document, window) {
     }
 
     if(topic != null) {
-        await fetch("/search?topic="+topic).then(response => response.json()).then((results) => {
-            var resultContainer = document.getElementById("result-container");
+        var tutors = getTutors(topic);
+        var books = getBooks(topic);
+        
+        await tutors;
+        await books;
```
Comment (betrevisan) [inline src/main/webapp/search.js:63]: What are these for?

File: src/main/java/com/google/sps/data/Book.java (hunk 2 above)
Comment (kashiish) [reply src/main/java/com/google/sps/data/Book.java:18]: Oh ya, forgot about that. Thanks!

File: src/main/webapp/search.js (hunk 3 above)
Comment (kashiish) [reply src/main/webapp/search.js:63]: They just make sure to wait for the functions, getTutors and getBooks, to return before redirectToResult returns.

**Pull request merged**